pages/api/token.ts

```diff
@@ -0,0 +1,52 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import { AccessToken } from 'livekit-server-sdk';
+import type { AccessTokenOptions, VideoGrant } from 'livekit-server-sdk';
+
+const apiKey = process.env.LK_API_KEY;
+const apiSecret = process.env.LK_API_SECRET;
+
+const createToken = (userInfo: AccessTokenOptions, grant: VideoGrant) => {
+  const at = new AccessToken(apiKey, apiSecret, userInfo);
+  at.addGrant(grant);
+  return at.toJwt();
+};
+
+export default async function tokenHandle(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  try {
+    const { roomName, identity, name } = req.query;
+
+    console.log(roomName);
+    console.log(identity);
+    console.log(name);
+
+    if (typeof identity !== 'string') {
+      throw Error('Provide one identity');
+    }
+    if (typeof roomName !== 'string') {
+      throw Error('Provide on Roomname');
+    }
+
+    if (Array.isArray(name)) {
+      throw Error('Provide a name');
+    }
+
+    const grant: VideoGrant = {
+      room: roomName,
+      roomJoin: true,
+      canPublish: true,
+      canSubscribe: true,
+    };
+
+    const token = createToken({ identity, name }, grant);
+    console.log(token);
+
+    res.status(200).json({ identity, accessToken: token });
+  } 
```

Comment:
Is it possible to avoid wrapping the full operation in a try/catch?
Ideally we only wrap areas that *can* go wrong in one so if something _does_ indeed go wrong, we can quickly and easily pinpoint what exactly is breaking.

Likewise, this should probably limit to only GET requests. As it stands right now, this will return stuff on any request which isn't a RESTful approach.